docs/hypervisors/aws.md

```diff
@@ -0,0 +1,143 @@
+above.  If not provided, Beaker will try to name an AMI Config using the host's
+platform string.
+
+The `type` references the type variable in the Amazon Image Config file as well,
+so this key picks out the particular AMI ID from the set available for this type
+of host platform.
+
+The `ami-size` variable refers to
+[instance types](https://aws.amazon.com/ec2/instance-types/) by their model name.
+Some examples of these values are "m3.large", "c4.xlarge", and "r3.8xlarge". The
+default value if this key is not provided used by Beaker is "m1.small".
+      
+
+# AWS Keys
+
+For any particular Beaker run, a new EC2 ssh key with a name of the form
+
+    Beaker-<username>-<sanitized_hostname>-<aws_keyname_modifier>-<readable_timestamp>
+
+will be created at the beginning of the run, & cleaned up at the end of the run.
+
+Everything up to `aws_keyname_modifier` will be the same if run from the same
+user on the same machine no matter when it's run. This means that if you're
+running from a CI system, all of these values will usually be the same, depending
+on your setup.
+
+`aws_keyname_modifier` will by default be a 10 digit random number string.
+`readable_timestamp`'s most fine grained unit is nanoseconds. Between the two of
+these, every Beaker run will generate a unique ssh key name.
+
+These keys are deleted automatically as a part of the cleanup process at the end
+of a Beaker run.
+
+# Zombie Killing
+
+If an EC2 host stays around after a Beaker run, we refer to it as a zombie :).
+Normal Beaker execution should not create zombies, but a common use case that
+can result in zombies is using the `--preserve-hosts` options.
+
+If you would like to be sure that you're not running up your EC2 bill via any
+leftover preserved hosts in your EC2 system, we recommend creating a zombie
+killing Beaker job.
+
+To setup a zombie killing job, you'll need a Beaker test that kills all the
+zombies (referred to later as `kill.rb`):
+
+    ec2 = AwsSdk.new( [], options )
+    ec2.kill_zombies( 0 )
+
+Refer to the
+[Rubydoc for the `kill_zombies` method]
+(http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/AwsSdk#kill_zombies-instance_method)
+to learn more about it's
+parameters. Running this should be as simple as this:
+
+    # beaker --tests kill.rb
+
+Note that the second argument is tested as a regex against key names, so you
+could use the key pattern described above to wipeout machines that match a
+pattern such as "Beaker-johnsmith", and it will catch all keys for the "johnsmith"
+user.
+
+### How Do I Find Out My Key Prefix?
+
+In order to find out your key pattern as used by Beaker, just kick off a Beaker
+run to generate an EC2 host. When you do this, you should see lines that look
+like so:
+
+    aws-sdk: Launch instance
+    aws-sdk: Ensure key pair exists, create if not
+    [AWS EC2 200 0.142666 0 retries]^[[0m^[[1m describe_key_pairs(:filters=>[{:name=>"key-name",:values=>["Beaker-johnsmith-Johns-Ubuntu-2-local"]}]) ^[[0
```

Comment:
It's so amazing to see hypervisor docs landing with any patch!

One little nit is the `^[[0m^[[1m`console escape sequences`^[[0` in here.